src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -17,17 +17,15 @@ namespace Akka.Persistence
 
     internal class EventsourcedState
     {
-        public EventsourcedState(string name, bool isRecoveryRunning, StateReceive stateReceive)
+        public EventsourcedState(string name, Func<bool> isRecoveryRunning, StateReceive stateReceive)
```

Comment:
Is this the best option?

Reply:
Question: how often is the `EventsourcedState` object allocated? From a functional standpoint it should be fine, but if we allocate a new `EventsourcedState` instance regularly then anonymous method allocations could add up. I'm not familiar enough with Akka.Persistence's internals to know off-hand. Do you or @Horusiath have some idea?

Reply:
@Aaronontheweb not sure about this one, honestly. Maybe @Horusiath wants to weigh in.